File: js/filter-list.js
/**
 * Tanggal: 7/3/2025
 * Link: https://www.codewars.com/kata/53dbd5315a3c69eed20002dd/train/javascript
 */

function filter_list(l) {
    let result = [];

    for (let item of l) {
        if (typeof item === "number") {
            result.push(item);
        }
    }

    return result;
}

console.log(filter_list([1, 2, "a", "b"]));
console.log(filter_list([1, "a", "b", 0, 15]));
console.log(filter_list([1, 2, "aasf", "1", "123", 123]));

File: java/FilterList.java
/**
 * Tanggal: 1 Mei 2025
 * Link: https://www.codewars.com/kata/53dbd5315a3c69eed20002dd/train/java
 */

import java.util.ArrayList;
import java.util.List;

public class FilterList {
    public static void main(String[] args) {
        System.out.println(filterList(List.of(1, 2, "a", "b")));
    }

    public static List<Object> filterList(final List<Object> list) {
        List<Object> result = new ArrayList<>();

        for (int i = 0; i < list.size(); i++) {
            if (list.get(i) instanceof Integer) {
                result.add(list.get(i));
            }
        }

        return result;
    }
}
